### ISSUE TITLE
Support for @2x, @3x, etc naming convention for images

### ISSUE THREAD

**Issue by jlubeck** [flutter/flutter#51953]
## Use case

When creating new images or reusing images from other projects, design apps export them with @2x, @3x or similar. Even if it is not possible to support having all these images in the same folder, at least being able to understand them when they are in the 2.0x, 3.0x folder would be great. 

## Proposal

When having an image.png file, and putting a higher quality image in the 2.0x folder, flutter should accept image@2x.png

**Comment by TahaTesser** [comment]
HI @jlubeck 
I believe this feature already exist
Please take a look at this [Declaring resolution-aware image assets
](https://flutter.dev/docs/development/ui/assets-and-images#resolution-aware) 

if you disagree please write in the comments and I will reopen it 
Thank you

**Comment by jlubeck** [comment]
Hi @TahaTesser , the model that I'm asking does not exist yet. I did mention in my original message about the 2.0x and 3.0x folders. But what I'm asking is to ALSO support the file names with the @2x and @3x suffixes. The reason being that most of the design softwares out there export the images in this format.

Thank you

**Comment by haliliceylan** [comment]
hi @jlubeck, i have same problem with Figma so I write little code snipet. maybe it can help you;

You need mmv package for move files based on regex

for MacOS `brew install mmv`
for Linux `sudo apt-get install mmv`

and make sure u have already 2.0x and 3.0x folder
`mkdir 2.0x`
`mkdir 3.0x`

with these commands move the files to correct folder.
( you can run multiple times don't worry)

```
mmv \*@2x\* 2.0x/\#1\#2
mmv \*@3x\* 3.0x/\#1\#2
```

**Comment by christopherfujino** [comment]
@andrewkolos does https://github.com/flutter/flutter/pull/110721 resolve this?

**Comment by andrewkolos** [comment]
> @andrewkolos does #110721 resolve this?

No. It was strictly for a fix for how we currently support resolution-aware image assets (by folder name). It doesn't add support for detecting higher resolution image variants from filenames alone.
